CMakeLists.txt

```diff
@@ -532,8 +532,8 @@ include("${CODING_CONV_CMAKE}/3rdparty.cmake")
 # =============================================================================
 # Add coding-conventions submodule if code formatting enabled
 # =============================================================================
-if(NEURON_CMAKE_FORMAT)
-  set(NEURON_CMakeFormat_EXCLUDES_RE
+if(NRN_CMAKE_FORMAT OR NRN_CLANG_FORMAT)
+  set(NRN_CMakeFormat_EXCLUDES_RE
```

Comment:
Oops, should have spotted that one. I think there is meant to be a feature in the coding conventions to not format submodules (`..._FORMATTING_NO_SUBMODULES`: https://github.com/BlueBrain/hpc-coding-conventions/blob/5dd9d8308937f9e9be7093cd579e4301dc43131c/cpp/README.md#code-formatting), but I never tried using it.

Reply:
by default, submodules are not formatted if I understand this right:

> Although it is possible to overwrite the default settings to restrict the scanned directories, the formatting applies to the entire repository except git submodules by default.
> 
> To enable the formatting of CMake and C++ code inside git submodules, enable the ${PROJECT_NAME}_FORMATTING_NO_SUBMODULES:BOOL CMake variable.

Reply:
Yeah...so I am wondering if we really need this `NRN_CMakeFormat_EXCLUDES_RE` thing at all. It looks like it was my fault: https://github.com/neuronsimulator/nrn/pull/1011/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a ... I wrote
> Discourage cmake-format from touching submodules (a better fix is presumably possible in https://github.com/BlueBrain/hpc-coding-conventions, but this improves the situation in the meantime)

in the PR description, but I don't currently see how that could have been true 🤷 😅

Reply:
> by default, submodules are not formatted if I understand this right:

I haven't verified this though...(might be depend on naming structure of submodule dir? `ext` vs `external`)